core/src/main/java/com/sequenceiq/cloudbreak/service/stack/CloudResourceAdvisor.java

```diff
@@ -106,7 +107,8 @@ public PlatformRecommendation createForBlueprint(Long workspaceId, String bluepr
         }
         VmRecommendations recommendations = cloudParameterService.getRecommendation(cloudPlatform);
 
-        Set<VmType> availableVmTypes = vmTypes.getCloudVmResponses().get(availabilityZone);
+        Set<VmType> availableVmTypes = StringUtils.isNotBlank(availabilityZone) ? vmTypes.getCloudVmResponses().get(availabilityZone)
+                : vmTypes.getCloudVmResponses().values().iterator().next();
```

Comment:
Instead of null you might get a NoSuchElementException from iterator().next() method call